demisto_sdk/commands/unify/tests/generic_module_unifier_test.py

```diff
@@ -0,0 +1,206 @@
+    dashboard_copy['id'] = 'wrong_id'
+    pack.create_dashboard('dashboard_2', dashboard_copy)
+    unifier = GenericModuleUnifier(input=generic_module_path)
+    found_dashboard = unifier.find_dashboard_by_id(dashboard_id=id_to_test)
+    assert found_dashboard is None
+
+
+def test_merge_generic_module_with_its_dashboards_positive(repo):
+    """
+    Given
+    - A pack with a valid generic module, and a dashboard that it's id matches a dashboard in the generic module.
+
+    When
+    - Running GenericModuleUnifier.merge_generic_module_with_its_dashboards()
+
+    Then
+    - Ensure the module was unified successfully - i.e contains the dashboard's content
+    """
+    pack = repo.create_pack('PackName')
+    pack.create_generic_module("generic-module", GENERIC_MODULE)
+    generic_module_path = pack.generic_modules[0].path
+    dashboard_copy = DASHBOARD.copy()
+    dashboard_copy['id'] = 'asset_dashboard'
+    pack.create_dashboard('dashboard_1', dashboard_copy)
+    unifier = GenericModuleUnifier(input=generic_module_path)
+    unified_generic_module = unifier.merge_generic_module_with_its_dashboards()
+    assert unified_generic_module == UNIFIED_GENERIC_MODULE
+
+
+def test_merge_generic_module_with_its_dashboards_negative(repo, capsys):
+    """
+    Given
+    - A pack with a valid generic module, and no dashboard that it's id matches a dashboard in the generic module.
+
+    When
+    - Running GenericModuleUnifier.merge_generic_module_with_its_dashboards()
+
+    Then
+    - Ensure the module wasn't unified.
+    - Ensure a suitable error message was printed.
+    """
+    pack = repo.create_pack('PackName')
+    pack.create_generic_module("generic-module", GENERIC_MODULE)
+    generic_module_path = pack.generic_modules[0].path
+    generic_module_dash_id = GENERIC_MODULE.get('views', {})[0].get('tabs', {})[0].get('dashboard', {}).get('id')
+    pack.create_dashboard('dashboard_1', DASHBOARD)
+    unifier = GenericModuleUnifier(input=generic_module_path)
+    non_unified_generic_module = unifier.merge_generic_module_with_its_dashboards()
+    assert non_unified_generic_module == GENERIC_MODULE
+    err_msg = capsys.readouterr()
+    assert f'Dashboard {generic_module_dash_id} was not found in pack: PackName and therefore was not unified\n' in\
+           err_msg
+
+
+def test_save_unified_generic_module(repo):
+    """
+    Given
+    - A unified generic module, and a desirable saving path.
+
+    When
+    - Running GenericModuleUnifier.save_unified_generic_module()
+
+    Then
+    - Ensure the module was saved successfully in the desirable path.
+    """
+    pack = repo.create_pack('PackName')
+    pack.create_generic_module("generic-module", GENERIC_MODULE)
+    generic_module_path = pack.generic_modules[0].path
+    unifier = GenericModuleUnifier(input=generic_module_path, output=pack._dashboards_path)
+    unifier.save_unified_generic_module(unified_generic_module_json=UNIFIED_GENERIC_MODULE)
```

Comment:
you are creating a new file here, use `with ChangeCWD(pack.repo_path):` to make sure the file is deleted after the unit test is done.
same for subsequent unit tests which use the same logic